internal/core/core.go

```diff
@@ -102,6 +102,10 @@ func BuildRootCmd(conf *config.Config) cobra.Command {
 		&conf.GlobalFlags.Debug, "debug", false,
 		"Print out more verbose debug logs.",
 	)
+	flags.BoolVar(
+		&conf.GlobalFlags.NoKeyring, "no-keyring", false,
+		"Disable system keyring integration for credential storage.",
+	)
```

Comment:
_Comment from #643 to make it visible for other reviewers:_

> There should be no need for `--no-keyring` flag: keyring has least priority when parsing the credentials, so with `--no-keyring` user would just get a slightly different error message. I.e., if credentials are defined in environment variables, the keyring should not be accessed. Maybe instead of adding `--no-keyring` flag the keyring access failed error message could be clarified.
>
> For `login --with-token`, the `--no-keyring` flag would not make sense as we do not currently have any alternative methods for saving the token. One alternative would be to save the token to upctl configuration file, but in that case the parameter should be about allowing this less secure fallback.